urld829b80034_wjch111_PTSP2013Wang_tgz-pJ8-controllers_momcts_momctsCore_TestPlannerJ8/src/wox/serial/Easy.java

```diff
@@ -0,0 +1,47 @@
+package wox.serial;
+
+import org.jdom.Element;
+import org.jdom.Document;
+import org.jdom.input.SAXBuilder;
+import org.jdom.output.XMLOutputter;
+
+import java.io.FileWriter;
+import java.io.InputStream;
+import java.io.FileInputStream;
+
+/**
+ * Created by IntelliJ IDEA.
+ * User: sml
+ * Date: 08-Nov-2004
+ * Time: 15:02:11
+ * To change this template use Options | File Templates.
+ */
+public class Easy {
+    public static void save(Object ob, String filename) {
+        try {
+            ObjectWriter writer = new SimpleWriter();
+            Element el = writer.write(ob);
+            XMLOutputter out = new XMLOutputter(); // ("  ", true);
+            FileWriter file = new FileWriter(filename);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *file*. The resource is referenced at line 26. The resource is closed at line 27. There are other execution paths that do not contain closure statements, for example, when XMLOutputter.output throws an exception. Ensure *file* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).